services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/configuration/ApiDisabledConfiguration.kt

```diff
@@ -0,0 +1,56 @@
+package fi.hel.haitaton.hanke.configuration
+
+import jakarta.servlet.Filter
+import jakarta.servlet.http.HttpServletResponse
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.boot.web.servlet.FilterRegistrationBean
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.core.annotation.Order
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.web.SecurityFilterChain
+
+@Configuration
+@ConditionalOnProperty(
+    name = ["haitaton.api.enabled"],
+    havingValue = "false",
+    matchIfMissing = false
+)
+class ApiDisabledConfiguration {
+
+    companion object {
+        val BLOCKED_PATHS =
```

Comment:
It would be better to whitelist the allowed paths. I don't expect anyone to remember to update this list when adding new endpoints.

Reply:
I did that. Even though when adding new endpoints the blocked/not-blocked check has to be done anyway.